backend/src/api/model/realm/mutations.rs

```diff
@@ -8,13 +8,20 @@ use crate::{
 use super::{Realm, RealmOrder};
 
 
+
 impl Realm {
     pub(crate) async fn add(realm: NewRealm, context: &Context) -> ApiResult<Realm> {
         let db = context.db(context.require_moderator()?);
 
+        // Check if the path is a reserved one.
+        let parent_key = id_to_key(realm.parent, "`parent`")?;
+        let path_is_reserved = context.config.general.reserved_paths()
```

Comment:
Why even run this check when `parent_key.0 != 0`? :thinking:

Reply:
The compiler is free to reorder this to only check for reserved path if necessary. Putting the condition in the if is too long and results in a 2 line if-condition, which is bad for readability. Or one could have another if nested, also not great.